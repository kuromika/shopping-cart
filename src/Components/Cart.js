import React from "react";
import CartItem from "./CartItem";

const Cart = (props) => {
    return (
        <div className="cart">
            <h1>Your Cart</h1>
            <div className="items" role='list'>
                {props.items.length === 0 ? <h2> Add items to your cart</h2> :
                    props.items.map( (item) => 
                        <CartItem
                            key={item.id}
                            img={item.image}
                            desc={item.desc}
                            name={item.title}
                            price={item.price}
                            id = {item.id}
                            quantity={item.quantity}
                            add={props.add}
                            remove={props.rm}
                        ></CartItem>)
                }
            </div>

            <div className="checkout">
                <div className="cart-total">
                    <span>Total: </span>
                    <span aria-label="total">{props.total.toFixed(2)}</span>
                </div>
            </div>
        </div>
    )
}

export default Cart;