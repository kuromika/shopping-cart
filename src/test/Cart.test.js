import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'
import Cart from "../Components/Cart";

//setup

const items = [
    { title: 'first item' },
    { title: 'second item' },
    { title: 'third item' }
];

//tests

describe('Cart is rendered properly', () => {
    
    test('Renders "Add items to your cart" when no items are passed', () => {
        const { container } = render(
        <Cart
            items={[]}
            total = {0}
        />);
        const heading = screen.getByRole('heading', {level:2});
        expect(heading.textContent).toMatch(/Add items to your cart/i);
        expect(container).toMatchSnapshot();
    })

    test('Renders the right amount of items', () => {

        render(<Cart
            items={items}
            total = {0}
        />)

        const listItems = screen.getAllByRole('listitem');
        expect(listItems.length).toBe(3);
    });

    test('Renders the right total with exactly 2 decimals', () => {

        render(<Cart
            items={[]}
            total={123.12345}
        />);

        const totalSpan = screen.getByLabelText('total');
        expect(totalSpan.textContent.length).toBe(6);
    })

})