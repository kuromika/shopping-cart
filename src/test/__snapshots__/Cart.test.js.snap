// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart is rendered properly Renders "Add items to your cart" when no items are passed 1`] = `
<div>
  <div
    class="cart"
  >
    <h1>
      Your Cart
    </h1>
    <div
      class="items"
      role="list"
    >
      <h2>
         Add items to your cart
      </h2>
    </div>
    <div
      class="checkout"
    >
      <div
        class="cart-total"
      >
        <span>
          Total: 
        </span>
        <span
          aria-label="total"
        >
          0.00
        </span>
      </div>
    </div>
  </div>
</div>
`;
